src/views/Entries/QualityControl/EntryCard/index.tsx

```diff
@@ -90,32 +99,64 @@ function EntryCard(props: EntryCardProps) {
     const {
         className,
         entry: entryFromProps,
-        lead,
+        lead: leadFromProps,
         framework,
         onDelete,
         isDeleted,
     } = props;
 
+    const [lead, setLead] = React.useState<Pick<Lead, EntryLeadType>>(leadFromProps);
+    const [isEditLeadModalShown, showEditLeadModal] = React.useState<boolean>(false);
+    const [entry, setEntry] = React.useState<Entry>(entryFromProps);
+    const [isVerified, setVerificationStatus] = React.useState<boolean>(entry.verified);
+
     const {
         url: leadUrlFromProps,
         attachment,
     } = lead;
 
     const leadUrl = (attachment && attachment.file) ?? leadUrlFromProps;
 
-    const leadSource = lead.sourceDetails ? lead.sourceDetails.title : lead.sourceRaw;
+    const leadSource = lead.sourceDetail ? lead.sourceDetail.title : lead.sourceRaw;
+
+    const [
+        pending,
+        leadFromRequest,
+        ,
+        getLead,
+    ] = useRequest<Lead>({
+        url: `server://v2/leads/${lead.id}`,
+        method: 'GET',
+    });
 
-    const handleDeletePendingChange = React.useCallback((/* isPending: boolean */) => {
+    useEffect(() => {
+        if(leadFromRequest) {
+            showEditLeadModal(true);
+        }
+    },[leadFromRequest]);
+
+    const handleDeletePendingChange = useCallback((/* isPending: boolean */) => {
         // TODO; disable all actions if pending
     }, []);
 
-    const [entry, setEntry] = React.useState<Entry>(entryFromProps);
-    const [isVerified, setVerificationStatus] = React.useState<boolean>(entry.verified);
-
-    const handleDeleteSuccess = React.useCallback(() => {
+    const handleDeleteSuccess = useCallback(() => {
         onDelete(entry.id);
     }, [onDelete, entry]);
 
+    const handleEditLeadButtonClick = () => {
+        getLead();
+    };
+
+    const handleEditLeadModalClose = () => {
+        showEditLeadModal(false);
+    };
+
+    const handleEntryEditSave = useCallback((lead: Lead) => {
+        setLead(lead);
```

Comment:
I think we should modify the data source instead of doing this only in this component?
Thoughts @samshara , @tnagorra ?

Reply:
patching the lead data can be done but I think it won't be useful here as when switching to the leads page the data is again pulled from the server. @tnagorra any thoughts?

Reply:
we shouldn't create a state for lead on individual card.
A change on lead inside a card should reflect that change to other entries card with the same lead

Reply:
@samshara , we might need to look for other entries with the same lead and change the data there as well.